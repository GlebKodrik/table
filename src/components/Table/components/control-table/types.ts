import { TSelectItem } from '../select/types';

export type TControlTable = Partial<{
  isDisabled: boolean;
  text: string;
  selectItems: TSelectItem[];
  position: 'center' | 'right' | 'left';
  totalCount: number;
  onPaginationChange: (page: number) => void;
  selectChange: (select: TSelectItem) => void;
  totalPages: number;
  currentPage: number;
  className: string;
  selectValue: TSelectItem;
}>;
export type TProps = TControlTable;
