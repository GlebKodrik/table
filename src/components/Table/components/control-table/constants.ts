export type TSelectItem = {
  label: string;
  id: number;
  value: number | null;
};

export const SELECT_ITEMS: TSelectItem[] = [
  { id: 1, label: '50', value: 50 },
  { id: 2, label: '100', value: 100 },
  { id: 3, label: '200', value: 200 },
  { id: 4, label: '400', value: 400 },
  { id: 5, label: '800', value: 800 },
];
