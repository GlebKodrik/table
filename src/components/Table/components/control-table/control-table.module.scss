.wrapper {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  min-height: 48px;
  padding: 8px 24px;
  background-color: white;
  border: 1px solid  var(--color-bg-border);
  border-radius: var(--control-radius);

  &-with-select {
    display: flex;
    justify-content: space-between;
  }

  &-select {
    display: flex;
    align-items: center;
  }
}

.total-count {
  flex: none;
  margin-left: auto;
}

.select {
  width: 100px;
  margin-left: 8px;
}

.disabled-select {
  cursor: not-allowed;
}

.last-element {
  display: flex;
  align-items: center;
  text-align: right;
}

.pagination {
  align-items: center;
}
