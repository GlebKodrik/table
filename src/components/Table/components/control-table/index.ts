export { ControlTable } from './control-table';
