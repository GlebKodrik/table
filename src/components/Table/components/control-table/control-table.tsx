import { Text } from '@consta/uikit/Text';
import cn from 'classnames';
import React, { memo, useContext } from 'react';

import { TableProvider } from '@/components/Table/providers/table-provider';

import { Pagination } from './components/pagination';
import { Select } from './components/select';
import { TotalCount } from './components/total-count';
import { SELECT_ITEMS } from './constants';
import styles from './control-table.module.scss';

export const ControlTable = memo(() => {
  const {
    perPage: propPerPage,
    onPerPageChange,
    perPageItems = SELECT_ITEMS,
    totalCount,
    page = 0,
    isDisabledControlMenu,
    onPaginationChange,
  } = useContext(TableProvider);
  const perPage = propPerPage || perPageItems[0];

  const isHaveSelect = perPage && onPerPageChange && perPageItems;
  const isHavePagination = Number.isFinite(page) && onPaginationChange;

  const totalPages = perPage?.value
    ? Math.ceil((totalCount || 0) / (perPage.value || 0))
    : null;

  if (!isHaveSelect && !isHavePagination && !totalCount) {
    return null;
  }

  return (
    <div
      className={cn(styles.wrapper, {
        [styles.wrapperWithSelect]: perPage || (totalPages && totalPages <= 1),
        [styles.lastElement]: !isHavePagination && !isHaveSelect,
      })}
    >
      {isHaveSelect ? (
        <div className={styles.wrapperSelect}>
          <>
            <Text>Показывать&nbsp;строк</Text>
            <div className={styles.select}>
              <Select
                className={cn({
                  [styles.disabledSelect]: isDisabledControlMenu,
                })}
                disabled={isDisabledControlMenu}
                items={perPageItems}
                onChange={onPerPageChange}
                value={perPage}
              />
            </div>
          </>
        </div>
      ) : null}
      {isHavePagination ? (
        <Pagination
          currentPage={page as number}
          totalPages={totalPages || 0}
          onChange={onPaginationChange}
          position="center"
          isDisabled={isDisabledControlMenu}
          className={styles.pagination}
        />
      ) : null}
      {totalCount ? (
        <TotalCount
          text="Всего элементов"
          totalCount={totalCount}
          className={styles.totalCount}
        />
      ) : null}
    </div>
  );
});
