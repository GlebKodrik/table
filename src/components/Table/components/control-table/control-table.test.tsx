import { render } from '@testing-library/react';

import { ControlTable } from './control-table';

describe('Component ControlTable', () => {
  test('ControlTable render without parameters', () => {
    const { asFragment } = render(<ControlTable />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('ControlTable children render with parameters', () => {
    const { asFragment } = render(
      <ControlTable
        selectChange={() => {}}
        onPaginationChange={() => {}}
        totalCount={0}
        selectItems={[
          { label: '20', value: 20, id: 1 },
          { label: '30', value: 40, id: 2 },
          { label: '60', value: 60, id: 2 },
        ]}
        selectValue={{ label: '30', value: 40, id: 2 }}
        totalPages={1000}
        currentPage={6}
        text="Всего количество"
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('ControlTable render with only totalCount', () => {
    const { asFragment } = render(
      <ControlTable
        totalCount={0}
        text="Всего количество"
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('ControlTable render with only pagination', () => {
    const { asFragment } = render(
      <ControlTable
        totalPages={1000}
        currentPage={6}
        onPaginationChange={() => {}}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('ControlTable render with only select', () => {
    const { asFragment } = render(
      <ControlTable
        selectItems={[
          { label: '20', value: 20, id: 1 },
          { label: '30', value: 40, id: 2 },
          { label: '60', value: 60, id: 2 },
        ]}
        selectValue={{ label: '30', value: 40, id: 2 }}
        selectChange={() => {}}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('ControlTable render with select and totalCount', () => {
    const { asFragment } = render(
      <ControlTable
        selectItems={[
          { label: '20', value: 20, id: 1 },
          { label: '30', value: 40, id: 2 },
          { label: '60', value: 60, id: 2 },
        ]}
        selectValue={{ label: '30', value: 40, id: 2 }}
        totalCount={0}
        text="Всего количество"
        selectChange={() => {}}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('ControlTable render with pagination and totalCount', () => {
    const { asFragment } = render(
      <ControlTable
        totalPages={1000}
        currentPage={6}
        totalCount={0}
        text="Всего количество"
        onPaginationChange={() => {}}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('ControlTable render with isDisabled', () => {
    const { asFragment } = render(
      <ControlTable
        selectItems={[
          { label: '20', value: 20, id: 1 },
          { label: '30', value: 40, id: 2 },
          { label: '60', value: 60, id: 2 },
        ]}
        selectChange={() => {}}
        selectValue={{ label: '30', value: 40, id: 2 }}
        totalPages={1000}
        currentPage={6}
        totalCount={0}
        text="Всего количество"
        onPaginationChange={() => {}}
        isDisabled
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('ControlTable render with pagination zero page', () => {
    const { asFragment } = render(
      <ControlTable
        onPaginationChange={() => {}}
        totalPages={1000}
        currentPage={0}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
