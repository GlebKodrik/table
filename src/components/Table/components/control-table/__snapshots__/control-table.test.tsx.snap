// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ControlTable ControlTable children render with parameters 1`] = `
<DocumentFragment>
  <div
    class="wrapper wrapperWithSelect"
  >
    <div
      class="wrapperSelect"
    >
      <div
        class="Text Text_lineHeight_m Text_size_m Text_view_primary"
      >
        Показывать строк
      </div>
      <div
        class="select"
      >
        <div
          class="Select Select_labelPosition_top Select_size_s Select_view_default Select_type_select"
        >
          <div
            class="Select-Body"
          >
            <div
              class="Select-SelectContainer Select-SelectContainer_view_default Select-SelectContainer_form_default"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                class="Select-Control"
              >
                <div
                  class="Select-ControlInner"
                >
                  <div
                    class="Select-ControlValueContainer"
                  >
                    <input
                      class="Select-FakeField"
                      readonly=""
                      type="button"
                    />
                    <span
                      class="Select-ControlValue"
                      title="30"
                    >
                      30
                    </span>
                  </div>
                </div>
                <span
                  class="Select-Indicators"
                >
                  <button
                    class="Select-IndicatorsDropdown"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="icons--Icon icons--Icon_size_s icons--Icon_color_mono IconSelect Select-DropdownIndicatorIcon"
                    >
                      <svg
                        class="icons--Icon-Svg"
                        viewBox="0 0 16 16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3.5 5 8 10l4.5-5h-9z"
                        />
                      </svg>
                    </span>
                  </button>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <nav
      class="Pagination Pagination_form_default Pagination_size_s Pagination_position_center"
    >
      <div
        class="Pagination-PrevPage"
      >
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default Button_withIcon MixFocus Pagination-ItemLeft Pagination-ItemLeft_show"
          type="button"
        >
          <span
            class="icons--Icon icons--Icon_size_xs icons--Icon_color_mono IconBackward Button-Icon Button-Icon_position_left"
          >
            <svg
              class="icons--Icon-Svg"
              viewBox="0 0 12 12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m2.71 6 2.647-2.646-.707-.708L.796 6.5l3.854 3.854.707-.708L2.711 7H10V6H2.71z"
              />
            </svg>
          </span>
          <span
            class="Button-Label"
          >
            Назад
          </span>
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_xs Text_view_ghost Pagination-TipLeft"
        />
      </div>
      <div
        class="Pagination-Pages"
      >
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="1"
          type="button"
        >
          1
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_s Text_view_linkMinor Pagination-More"
        >
          ...
        </div>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="5"
          type="button"
        >
          5
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="6"
          type="button"
        >
          6
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item Pagination-Item_isActive"
          title="7"
          type="button"
        >
          7
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="8"
          type="button"
        >
          8
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="9"
          type="button"
        >
          9
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="10"
          type="button"
        >
          10
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_s Text_view_linkMinor Pagination-More"
        >
          ...
        </div>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="1000"
          type="button"
        >
          1000
        </button>
      </div>
      <div
        class="Pagination-NextPage"
      >
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default Button_withIcon MixFocus Pagination-ItemRight Pagination-ItemRight_show"
          type="button"
        >
          <span
            class="Button-Label"
          >
            Вперёд
          </span>
          <span
            class="icons--Icon icons--Icon_size_xs icons--Icon_color_mono IconForward Button-Icon Button-Icon_position_right"
          >
            <svg
              class="icons--Icon-Svg"
              viewBox="0 0 12 12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.293 7H1V6h7.293L5.646 3.354l.708-.708L10.207 6.5l-3.853 3.854-.708-.708L8.293 7z"
              />
            </svg>
          </span>
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_xs Text_view_ghost Pagination-TipRight"
        />
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Component ControlTable ControlTable render with isDisabled 1`] = `
<DocumentFragment>
  <div
    class="wrapper wrapperWithSelect"
  >
    <div
      class="wrapperSelect"
    >
      <div
        class="Text Text_lineHeight_m Text_size_m Text_view_primary"
      >
        Показывать строк
      </div>
      <div
        class="select"
      >
        <div
          class="Select Select_labelPosition_top Select_size_s Select_view_default Select_type_select disabledSelect"
        >
          <div
            class="Select-Body"
          >
            <div
              class="Select-SelectContainer Select-SelectContainer_view_default Select-SelectContainer_form_default Select-SelectContainer_disabled"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                class="Select-Control"
              >
                <div
                  class="Select-ControlInner"
                >
                  <div
                    class="Select-ControlValueContainer"
                  >
                    <input
                      class="Select-FakeField"
                      readonly=""
                      type="button"
                    />
                    <span
                      class="Select-ControlValue"
                      title="30"
                    >
                      30
                    </span>
                  </div>
                </div>
                <span
                  class="Select-Indicators"
                >
                  <button
                    class="Select-IndicatorsDropdown"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="icons--Icon icons--Icon_size_s icons--Icon_color_mono IconSelect Select-DropdownIndicatorIcon"
                    >
                      <svg
                        class="icons--Icon-Svg"
                        viewBox="0 0 16 16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3.5 5 8 10l4.5-5h-9z"
                        />
                      </svg>
                    </span>
                  </button>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <nav
      class="Pagination Pagination_form_default Pagination_size_s Pagination_position_center disabled"
    >
      <div
        class="Pagination-PrevPage"
      >
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default Button_withIcon MixFocus Pagination-ItemLeft Pagination-ItemLeft_show"
          type="button"
        >
          <span
            class="icons--Icon icons--Icon_size_xs icons--Icon_color_mono IconBackward Button-Icon Button-Icon_position_left"
          >
            <svg
              class="icons--Icon-Svg"
              viewBox="0 0 12 12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m2.71 6 2.647-2.646-.707-.708L.796 6.5l3.854 3.854.707-.708L2.711 7H10V6H2.71z"
              />
            </svg>
          </span>
          <span
            class="Button-Label"
          >
            Назад
          </span>
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_xs Text_view_ghost Pagination-TipLeft"
        />
      </div>
      <div
        class="Pagination-Pages"
      >
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="1"
          type="button"
        >
          1
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_s Text_view_linkMinor Pagination-More"
        >
          ...
        </div>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="5"
          type="button"
        >
          5
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="6"
          type="button"
        >
          6
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item Pagination-Item_isActive"
          title="7"
          type="button"
        >
          7
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="8"
          type="button"
        >
          8
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="9"
          type="button"
        >
          9
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="10"
          type="button"
        >
          10
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_s Text_view_linkMinor Pagination-More"
        >
          ...
        </div>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="1000"
          type="button"
        >
          1000
        </button>
      </div>
      <div
        class="Pagination-NextPage"
      >
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default Button_withIcon MixFocus Pagination-ItemRight Pagination-ItemRight_show"
          type="button"
        >
          <span
            class="Button-Label"
          >
            Вперёд
          </span>
          <span
            class="icons--Icon icons--Icon_size_xs icons--Icon_color_mono IconForward Button-Icon Button-Icon_position_right"
          >
            <svg
              class="icons--Icon-Svg"
              viewBox="0 0 12 12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.293 7H1V6h7.293L5.646 3.354l.708-.708L10.207 6.5l-3.853 3.854-.708-.708L8.293 7z"
              />
            </svg>
          </span>
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_xs Text_view_ghost Pagination-TipRight"
        />
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Component ControlTable ControlTable render with only pagination 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <nav
      class="Pagination Pagination_form_default Pagination_size_s Pagination_position_center"
    >
      <div
        class="Pagination-PrevPage"
      >
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default Button_withIcon MixFocus Pagination-ItemLeft Pagination-ItemLeft_show"
          type="button"
        >
          <span
            class="icons--Icon icons--Icon_size_xs icons--Icon_color_mono IconBackward Button-Icon Button-Icon_position_left"
          >
            <svg
              class="icons--Icon-Svg"
              viewBox="0 0 12 12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m2.71 6 2.647-2.646-.707-.708L.796 6.5l3.854 3.854.707-.708L2.711 7H10V6H2.71z"
              />
            </svg>
          </span>
          <span
            class="Button-Label"
          >
            Назад
          </span>
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_xs Text_view_ghost Pagination-TipLeft"
        />
      </div>
      <div
        class="Pagination-Pages"
      >
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="1"
          type="button"
        >
          1
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_s Text_view_linkMinor Pagination-More"
        >
          ...
        </div>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="5"
          type="button"
        >
          5
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="6"
          type="button"
        >
          6
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item Pagination-Item_isActive"
          title="7"
          type="button"
        >
          7
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="8"
          type="button"
        >
          8
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="9"
          type="button"
        >
          9
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="10"
          type="button"
        >
          10
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_s Text_view_linkMinor Pagination-More"
        >
          ...
        </div>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="1000"
          type="button"
        >
          1000
        </button>
      </div>
      <div
        class="Pagination-NextPage"
      >
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default Button_withIcon MixFocus Pagination-ItemRight Pagination-ItemRight_show"
          type="button"
        >
          <span
            class="Button-Label"
          >
            Вперёд
          </span>
          <span
            class="icons--Icon icons--Icon_size_xs icons--Icon_color_mono IconForward Button-Icon Button-Icon_position_right"
          >
            <svg
              class="icons--Icon-Svg"
              viewBox="0 0 12 12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.293 7H1V6h7.293L5.646 3.354l.708-.708L10.207 6.5l-3.853 3.854-.708-.708L8.293 7z"
              />
            </svg>
          </span>
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_xs Text_view_ghost Pagination-TipRight"
        />
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Component ControlTable ControlTable render with only select 1`] = `
<DocumentFragment>
  <div
    class="wrapper wrapperWithSelect"
  >
    <div
      class="wrapperSelect"
    >
      <div
        class="Text Text_lineHeight_m Text_size_m Text_view_primary"
      >
        Показывать строк
      </div>
      <div
        class="select"
      >
        <div
          class="Select Select_labelPosition_top Select_size_s Select_view_default Select_type_select"
        >
          <div
            class="Select-Body"
          >
            <div
              class="Select-SelectContainer Select-SelectContainer_view_default Select-SelectContainer_form_default"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                class="Select-Control"
              >
                <div
                  class="Select-ControlInner"
                >
                  <div
                    class="Select-ControlValueContainer"
                  >
                    <input
                      class="Select-FakeField"
                      readonly=""
                      type="button"
                    />
                    <span
                      class="Select-ControlValue"
                      title="30"
                    >
                      30
                    </span>
                  </div>
                </div>
                <span
                  class="Select-Indicators"
                >
                  <button
                    class="Select-IndicatorsDropdown"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="icons--Icon icons--Icon_size_s icons--Icon_color_mono IconSelect Select-DropdownIndicatorIcon"
                    >
                      <svg
                        class="icons--Icon-Svg"
                        viewBox="0 0 16 16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3.5 5 8 10l4.5-5h-9z"
                        />
                      </svg>
                    </span>
                  </button>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Component ControlTable ControlTable render with only totalCount 1`] = `
<DocumentFragment>
  <div
    class="wrapper lastElement"
  />
</DocumentFragment>
`;

exports[`Component ControlTable ControlTable render with pagination and totalCount 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <nav
      class="Pagination Pagination_form_default Pagination_size_s Pagination_position_center"
    >
      <div
        class="Pagination-PrevPage"
      >
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default Button_withIcon MixFocus Pagination-ItemLeft Pagination-ItemLeft_show"
          type="button"
        >
          <span
            class="icons--Icon icons--Icon_size_xs icons--Icon_color_mono IconBackward Button-Icon Button-Icon_position_left"
          >
            <svg
              class="icons--Icon-Svg"
              viewBox="0 0 12 12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m2.71 6 2.647-2.646-.707-.708L.796 6.5l3.854 3.854.707-.708L2.711 7H10V6H2.71z"
              />
            </svg>
          </span>
          <span
            class="Button-Label"
          >
            Назад
          </span>
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_xs Text_view_ghost Pagination-TipLeft"
        />
      </div>
      <div
        class="Pagination-Pages"
      >
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="1"
          type="button"
        >
          1
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_s Text_view_linkMinor Pagination-More"
        >
          ...
        </div>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="5"
          type="button"
        >
          5
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="6"
          type="button"
        >
          6
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item Pagination-Item_isActive"
          title="7"
          type="button"
        >
          7
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="8"
          type="button"
        >
          8
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="9"
          type="button"
        >
          9
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="10"
          type="button"
        >
          10
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_s Text_view_linkMinor Pagination-More"
        >
          ...
        </div>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="1000"
          type="button"
        >
          1000
        </button>
      </div>
      <div
        class="Pagination-NextPage"
      >
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default Button_withIcon MixFocus Pagination-ItemRight Pagination-ItemRight_show"
          type="button"
        >
          <span
            class="Button-Label"
          >
            Вперёд
          </span>
          <span
            class="icons--Icon icons--Icon_size_xs icons--Icon_color_mono IconForward Button-Icon Button-Icon_position_right"
          >
            <svg
              class="icons--Icon-Svg"
              viewBox="0 0 12 12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.293 7H1V6h7.293L5.646 3.354l.708-.708L10.207 6.5l-3.853 3.854-.708-.708L8.293 7z"
              />
            </svg>
          </span>
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_xs Text_view_ghost Pagination-TipRight"
        />
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Component ControlTable ControlTable render with pagination zero page 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <nav
      class="Pagination Pagination_form_default Pagination_size_s Pagination_position_center"
    >
      <div
        class="Pagination-Pages"
      >
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item Pagination-Item_isActive"
          title="1"
          type="button"
        >
          1
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="2"
          type="button"
        >
          2
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="3"
          type="button"
        >
          3
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="4"
          type="button"
        >
          4
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="5"
          type="button"
        >
          5
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="6"
          type="button"
        >
          6
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="7"
          type="button"
        >
          7
        </button>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="8"
          type="button"
        >
          8
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_s Text_view_linkMinor Pagination-More"
        >
          ...
        </div>
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default MixFocus Pagination-Item"
          title="1000"
          type="button"
        >
          1000
        </button>
      </div>
      <div
        class="Pagination-NextPage"
      >
        <button
          class="Button Button_size_s Button_view_ghost Button_width_default Button_form_default Button_withIcon MixFocus Pagination-ItemRight Pagination-ItemRight_show"
          type="button"
        >
          <span
            class="Button-Label"
          >
            Вперёд
          </span>
          <span
            class="icons--Icon icons--Icon_size_xs icons--Icon_color_mono IconForward Button-Icon Button-Icon_position_right"
          >
            <svg
              class="icons--Icon-Svg"
              viewBox="0 0 12 12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.293 7H1V6h7.293L5.646 3.354l.708-.708L10.207 6.5l-3.853 3.854-.708-.708L8.293 7z"
              />
            </svg>
          </span>
        </button>
        <div
          class="Text Text_lineHeight_m Text_size_xs Text_view_ghost Pagination-TipRight"
        />
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Component ControlTable ControlTable render with select and totalCount 1`] = `
<DocumentFragment>
  <div
    class="wrapper wrapperWithSelect"
  >
    <div
      class="wrapperSelect"
    >
      <div
        class="Text Text_lineHeight_m Text_size_m Text_view_primary"
      >
        Показывать строк
      </div>
      <div
        class="select"
      >
        <div
          class="Select Select_labelPosition_top Select_size_s Select_view_default Select_type_select"
        >
          <div
            class="Select-Body"
          >
            <div
              class="Select-SelectContainer Select-SelectContainer_view_default Select-SelectContainer_form_default"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                class="Select-Control"
              >
                <div
                  class="Select-ControlInner"
                >
                  <div
                    class="Select-ControlValueContainer"
                  >
                    <input
                      class="Select-FakeField"
                      readonly=""
                      type="button"
                    />
                    <span
                      class="Select-ControlValue"
                      title="30"
                    >
                      30
                    </span>
                  </div>
                </div>
                <span
                  class="Select-Indicators"
                >
                  <button
                    class="Select-IndicatorsDropdown"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="icons--Icon icons--Icon_size_s icons--Icon_color_mono IconSelect Select-DropdownIndicatorIcon"
                    >
                      <svg
                        class="icons--Icon-Svg"
                        viewBox="0 0 16 16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3.5 5 8 10l4.5-5h-9z"
                        />
                      </svg>
                    </span>
                  </button>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Component ControlTable ControlTable render without parameters 1`] = `
<DocumentFragment>
  <div
    class="wrapper lastElement"
  />
</DocumentFragment>
`;
